modules/subscription/main.tf

```diff
@@ -1,36 +1,53 @@
 locals {
-  subscriptionPullLength = "${length(var.pullSubscriptions)}"
-  subscriptionPushLength = "${length(var.pushSubscriptions)}"
+  subscriptionPullLength = length(var.pullSubscriptions)
+  subscriptionPushLength = length(var.pushSubscriptions)
 }
 
 resource "google_pubsub_subscription" "pubsub_subscription_pull" {
-  count   = "${local.subscriptionPullLength}"
-  topic   = "${var.topicName}"
-  project = "${var.project}"
+  count   = local.subscriptionPullLength
+  topic   = var.topicName
+  project = var.project
 
-  name                 = "${lookup(var.pullSubscriptions[count.index], "name")}"
-  ack_deadline_seconds = "${lookup(var.pullSubscriptions[count.index], "deadline", 10)}"
-  message_retention_duration = "${lookup(var.pullSubscriptions[count.index], "retentionDuration", "604800")}s"
-  retain_acked_messages = "${lookup(var.pullSubscriptions[count.index], "retainAcked", false)}"
-  expiration_policy = {
-    ttl = "${lookup(var.pullSubscriptions[count.index], "expirationPolicy", "2678400")}s"
+  name                 = var.pullSubscriptions[count.index]["name"]
+  ack_deadline_seconds = lookup(var.pullSubscriptions[count.index], "deadline", 10)
+  message_retention_duration = "${lookup(
+    var.pullSubscriptions[count.index],
+    "retentionDuration",
+    "604800",
```

Comment:
I would make this not hardocded if possible?

Reply:
Do you mean by dynamic block?  We are hardcoding only a default value. Am I correct @rokerkony ?

Reply:
yes, those are only default values if none is provided

Reply:
👍